.github/workflows/code_formatting_checks.yml

```diff
@@ -0,0 +1,55 @@
+name: Code formatting
+
+# Scheduled at 11:30 PM UTC (05:00 AM IST)
+on:
+  schedule:
+    - cron: "30 23 * * *"
+
+jobs:
+  fix_code_formatting:
+    runs-on: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Fix code formatting issues
+        run: ./gradlew spotlessApply
+
+      - name: Commit and Push
+        run: ./scripts/commit_and_push.sh "${{secrets.USER_NAME}}" "${{secrets.USER_EMAIL}}"
+
+      - name: Create PR
+        uses: actions/github-script@0.3.0
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            var monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+            var date = new Date();
+            var monthName = monthNames[date.getMonth()];
+            var day = `${date.getDate()}`;
+            if (day.length < 2) {
```

Comment:
JavaScript has a [padStart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart) method that we might be able to use instead of this.